### ISSUE TITLE
loading multiple git configuration with bootstrap=true does not work

### ISSUE THREAD

**Issue by matheusgg** [spring-cloud/spring-cloud-config#1014]
**Spring Cloud Version Finchley.RC1**

**Problem:** After PR #912 simple configuration with only one git backend is not working. When config-server has its own configuration fetched from a remote repo (`spring.cloud.config.server.bootstrap=true`) it does not update repos map from class MultipleJGitEnvironmentRepository anymore.

**Sample project to re-create the issue:**
https://github.com/matheusgg/config-server-sample

**Remote repos for test:**
https://github.com/matheusgg/default-config
https://github.com/matheusgg/team1/tree/master
https://github.com/matheusgg/team2/tree/master

**Steps to reproduce:**
- Go to project directory and run `./gradlew clean build bootRun`
- According to the configuration file `config-server-sample.yml` (located at https://github.com/matheusgg/default-config), It should create 3 folders under ${user.home}/config-server-sample: **default**, **team1** and **team2**. But only **default** folder is being created.
- Finally, calling http://localhost:8888/team1-service-a/default/master or http://localhost:8888/team2-service-b/default/master should return the content of the files in **propertySources** property, but It doesn't.

**Comment by pivotal-dylan-roberts** [comment]
Hi @matheusgg 
If you don't mind, would you please provide steps reproduce?
I'm not clear on the precise issue you're experiencing.

Something like this:

Steps to reproduce:
1. start the app
2. GET endpoint /xyz
3. See response body

Expected value: "abc"
Actual value: "123"

**Comment by spencergibb** [comment]
So not quite a "simple" configuration as it has `spring.cloud.config.server.bootstrap: true`. The repos are loaded from a remote git repo.

Did this work in some non-milestone release prior?

**Comment by spencergibb** [comment]
looks like changing 
```
@ConditionalOnMissingBean
```
to
```
@ConditionalOnMissingBean(search = SearchStrategy.CURRENT)
```
on both the `EnvironmentRepository` and the property bean fixes this.

**Comment by spencergibb** [comment]
I've checked in a temporary fix to see if it fixes your problem. After the build, can you refresh your snapshots and try again?

**Comment by matheusgg** [comment]
Yes, sure! Thanks!

**Comment by matheusgg** [comment]
Hi @spencergibb I've tested here with your fix and it worked. Thank you!

**Comment by spencergibb** [comment]
This should have been closed in 2.0.0.RELEASE
